src/components/implementation/no_interface/new_boot/boot_deps.h

```diff
@@ -0,0 +1,147 @@
+#include <print.h>
+
+#undef assert
+#define assert(node) do { if (unlikely(!(node))) { debug_print("assert error in @ "); *((int *)0) = 0;} } while(0)
+
+#include <cos_alloc.h>
+#include <cobj_format.h>
+#include <cos_types.h>
+
+/*the booter uses this to keep track of each comp*/
+struct comp_cap_info {
+	struct cos_compinfo cos_compinfo;
+	vaddr_t addr_start;
+	vaddr_t vaddr_mapped_in_booter;
+	vaddr_t upcall_entry;
+};
+struct comp_cap_info comp_cap_info[MAX_NUM_SPDS+1];
+
+struct cos_compinfo boot_info;
+
+thdcap_t schedule[MAX_NUM_SPDS+1];
+unsigned int cur_sched;
+
+static int
+boot_comp_map_memory(struct cobj_header *h, spdid_t spdid, pgtblcap_t pt)
+{
+	int i, j;
+	int flag;
+	vaddr_t dest_daddr, prev_map = 0;
+	int tot = 0, n_pte = 1;	
+	
+	struct cobj_sect *sect = cobj_sect_get(h, 0);
+	
+	/*Expand Page table, could do this faster*/
+	for(j = 0; j < (int)h->nsect; j++){
+		tot += cobj_sect_size(h, j);
+	}
+	printc("tot: %d\n", tot);
+	if (tot > SERVICE_SIZE) {
+		n_pte = tot / SERVICE_SIZE;
+		if(tot % SERVICE_SIZE) n_pte++;
+	}	
+	for(j = 0; j < n_pte; j++){
+		if (!__bump_mem_expand_range(&boot_info, pt, sect->vaddr, SERVICE_SIZE)) BUG();
```

Comment:
I don't think using the private function from `cos_kernel_api.c` is right here. If you need that functionality, you probably need to make a public API in `cos_kernel_api.h` or make a private function in this component.